sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureSeekableByteChannel.java

```diff
@@ -0,0 +1,191 @@
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.NonReadableChannelException;
+import java.nio.channels.NonWritableChannelException;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileAttribute;
+
+/**
+ * A byte channel that maintains a current position.
+ * <p>
+ * This type is primarily offered to support some jdk convenience methods such as
+ * {@link Files#createFile(Path, FileAttribute[])} which requires opening a channel and closing it. A channel may only
+ * be opened in read mode OR write mode. It may not be opened in read/write mode. Seeking is supported for reads, but
+ * not for writes. Modifications to existing files is not permitted--only creating new files or overwriting existing
+ * files.
+ * <p>
+ * This type is not threadsafe to prevent having to hold locks across network calls.
+ * <p>
+ * {@link NioBlobInputStream} and {@link NioBlobOutputStream} are the preferred types for reading and writing blob data
+ * and are used internally by this type.
+ */
+public class AzureSeekableByteChannel implements SeekableByteChannel {
+    private final ClientLogger logger = new ClientLogger(AzureSeekableByteChannel.class);
+
+    private final NioBlobInputStream inputStream; // Always set the mark at 0 w/ max int. repositioning is a reset and then a skip. Validate length
+    private final NioBlobOutputStream outputStream;
+    private long position; // Needs to be threadsafe?
+    private boolean closed = false;
+    private final Path path;
+
+    AzureSeekableByteChannel(NioBlobInputStream inputStream, Path path) {
+        this.inputStream = inputStream;
+        inputStream.mark(Integer.MAX_VALUE);
+        this.outputStream = null;
+        this.position = 0;
+        this.path = path;
+    }
+
+    AzureSeekableByteChannel(NioBlobOutputStream outputStream, Path path) {
+        this.outputStream = outputStream;
+        this.inputStream = null;
+        this.position = 0;
+        this.path = path;
+    }
+
+    @Override
+    public int read(ByteBuffer dst) throws IOException {
+        AzurePath.ensureFileSystemOpen(this.path);
+        validateOpen();
+        validateReadMode();
+
+        // See comments in position()
+        if (this.position > this.size()) {
+            return -1;
+        }
+
+        int count = 0;
+
+        int len = dst.remaining();
+        byte[] buf = new byte[len];
+
+        while (count < len) {
+            int retCount = this.inputStream.read(buf, count, len - count);
+            if (retCount == -1) {
+                break;
+            }
+            count += retCount;
+        }
+        dst.put(buf, 0, count);
+        this.position += count;
+
+        return count;
+        // Would it be easier to just do the downloads directly?
```

Comment:
I think its probably better to just wrap BlobIS since it takes care of like buffering and stuff so we dont need to be issuing multiple reads for no reason.